src/server/routes/revision.js

```diff
@@ -168,17 +168,24 @@ function diffRevisionsWithParents(orm, entityRevisions, entityType) {
 		(revision) =>
 			revision.parent()
 				.then(
-					(parent) => makePromiseFromObject({
-						changes: revision.diff(parent),
-						entity: revision.related('entity'),
-						entityAlias: revision.get('dataId') ?
-							revision.related('data').fetch({require: false, withRelated: ['aliasSet.defaultAlias', 'aliasSet.aliases']}) :
-							orm.func.entity.getEntityParentAlias(
-								orm, entityType, revision.get('bbid')
-							),
-						isNew: !parent,
-						revision
-					}),
+					(parent) => {
+						const dataId = revision.get('dataId');
+						let isNew = false;
+						if (!parent) {
+							isNew = dataId;
+						}
+						return makePromiseFromObject({
+							changes: revision.diff(parent),
+							entity: revision.related('entity'),
+							entityAlias: revision.get('dataId') ?
```

Comment:
Since we've already extracted this as a constant:

```suggestion
							entityAlias: dataId ?
```

Reply:
> Great, that's a good improvement!
> 
> I think it would be great to take that opportunity to mark deleted entities with a `Deleted` badge with a red background, now that you've got that piece of info in your logic.
> I guess the cleanest would be to pass a new `isDeletion` property.
> 
> What do you think?